src/main/java/com/powsybl/openloadflow/network/LfNetwork.java

```diff
@@ -211,14 +211,14 @@ private void invalidateSlackAndReference() {
 
     public void updateSlackBusesAndReferenceBus() {
         if (slackBuses == null && referenceBus == null) {
-            SelectedSlackBus selectedSlackBus = slackBusSelector.select(busesByIndex, maxSlackBusCount);
-            slackBuses = selectedSlackBus.getBuses().stream()
-                    .filter(bus -> !excludedSlackBuses.contains(bus))
-                    .toList();
+            List<LfBus> selectableBus = busesByIndex.stream()
+                .filter(bus -> !excludedSlackBuses.contains(bus))
+                .toList();
+            SelectedSlackBus selectedSlackBus = slackBusSelector.select(selectableBus, maxSlackBusCount);
+            slackBuses = selectedSlackBus.getBuses();
             if (slackBuses.isEmpty()) { // ultimate fallback
-                selectedSlackBus = SLACK_BUS_SELECTOR_FALLBACK.select(busesByIndex, excludedSlackBuses.size() + maxSlackBusCount);
+                selectedSlackBus = SLACK_BUS_SELECTOR_FALLBACK.select(selectableBus, excludedSlackBuses.size() + maxSlackBusCount);
```

Comment:
Hello, a small remark here : There is no excluded slack buses in `selectableBus` so the limit can be modified to just `maxSlackBusCount` ?

Is there a use case that can cover this line ? Maybe using a NameSlackBusSelector ? It seems that Sonar wants more coverage on this file :)

Reply:
Yes you are right thanks. I am thinking of an example to test fallback mechanism right now.